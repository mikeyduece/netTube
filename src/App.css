.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}


// Boilerplate stuff

html, body {
  height: 100%;
}

body {
  background-color: #f0f0f0;
  display: flex;
  align-items: center;
  justify-content: center;
  background: url(https://assets.nflxext.com/ffe/siteui/vlv3/5e3955d1-a2b3-4753-bff2-875bf84330fa/412c42ad-ce5f-4c66-ad9f-3f63cab49b22/US-en-20171030-popsignuptwoweeks-perspective_alpha_website_small.jpg);
}

button ~ button {
  margin-left: 20px;
}

#login-page {
  margin: auto;
  text-align: center;
  padding-top: 20em;
}

.app-name {
  font-family: Nobile;
  color: red;
  text-shadow: .2rem .2rem .5rem white, .2rem .2rem .75rem black;
  font-weight: bolder;
  text-shadow: 0.25rem 0.25rem 0.75rem white, 0 5px black;
  font-weight: bolder;
  font-size: 10rem;
}
