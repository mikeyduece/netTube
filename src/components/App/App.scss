@import '../../stylesheets/base';

.app {
  height: inherit;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;

  h1 {
    color: $white;
    font-size: 15vw;
  }

  #logo {
    color: $vivid-red;
    line-height: normal;
    vertical-align: center;
  }
}