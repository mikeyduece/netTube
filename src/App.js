import React, { Component } from 'react';
import _ from 'lodash'
import GoogleLogin from 'react-google-login';
import { PostUser } from './components/PostUser'
import { endUserSession } from './components/endUserSession'
import './components/LoginButton/LoginButton'
import ENV from './config'
import Home from './components/Home/Home'
import './App.css';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      deets: null,
      videos: [],
      users: [],
      friendReqs: [],
      requests: []
    }
  }

  acceptFriend(e){
    let email = JSON.parse(localStorage.userData).email
    let friend_email = e.target.closest('div').children[0].textContent
    let tokenId = JSON.parse(localStorage.userData).token
    fetch('http://localhost:3000/api/v1/users/'+email+'/accept_request/'+friend_email,{
      method: 'PATCH',
      headers: {
        "HTTP_AUTHORIZATION": `${tokenId}`,
        'Authorization': tokenId,
        'Content-Type': 'application/json',
        'Accept': 'application/json'
      },
      mode: 'cors',
    })
  }

  handleSearch(video) {
    fetch(`http://localhost:3000/api/v1/search?q=${video}`)
      .then(response => response.json())
      .then((data) => {
        let compacted = _.compact(data)
        this.parseVideos(compacted)
      })
      .catch((err) =>{
        console.log(err)
      })
  }

  parseVideos(data) {
    this.setState({videos: data})
  }

  getUsers() {
    let email =''
    if(localStorage.userData !== undefined){
      email = JSON.parse(localStorage.userData).email
    }
    fetch('http://localhost:3000/api/v1/users/'+email+'/all_users')
    .then(response => response.json())
    .then(data => {
      localStorage.setItem('users', JSON.stringify(data))
    })
    .catch(err => {
      console.log(err)
    })
  }

  parseUsers(data) {
    this.setState({users: data})
  }

  getFriendReqs() {
    let email =''
    if(localStorage.userData !== undefined){
      email = JSON.parse(localStorage.userData).email
    }
    fetch('http://localhost:3000/api/v1/users/'+email+'/requests')
    .then(resp => resp.json())
    .then(data => {
      localStorage.setItem('friendReqs', JSON.stringify(data))
    })
  }

  componentWillMount(){
    this.getUsers()
  }

  handleIncomingFriendReq(){
    let users = ''
    let incoming = ''
    if(localStorage.friendReqs !== undefined &&
        localStorage.friendReqs.length !== 0 &&
        localStorage.userData !== undefined){
      users = JSON.parse(localStorage.users)
      incoming = JSON.parse(localStorage.friendReqs).incoming
      users.filter(user => {
       incoming.map(req => {
        if(user.id === req.user_id){
          this.state.requests.push(user)
        }
      })
    })
    }
  }

  componentDidMount(){
    this.getFriendReqs()
    this.handleIncomingFriendReq()
  }

  signIn(res){
    PostUser(res, res.accessToken)
    .then(data => {
      let userData = {
        'id': data.id,
        'full_name': data.name,
        'first_name': data.first_name,
        'last_name': data.last_name,
        'email': data.email,
        'token': data.token,
        'image': data.image
      }
      if(userData.token === res.accessToken){
        localStorage.setItem('userData', JSON.stringify(userData))
        this.setState({deets: JSON.parse(localStorage.userData)})
      }
    })
  }


  renderLogin() {
    const responseGoogle = (response) => {
      console.log(response)
      this.signIn(response)
    }

    return(
      <div id='login-page'>
      <h1 className='app-name'>netTUBE</h1>
      <GoogleLogin
      className='loginBtn loginBtn--google'
      clientId={ENV['GOOGLE_ID']}
      onSuccess={responseGoogle}
      onFailure={responseGoogle}
      />
      </div>
    )
  }

  logout(){
    console.log('logout')
    endUserSession(this.state.deets)
    this.setState({deets: null})
    localStorage.removeItem('userData')
  }

  loginOrHome(){
    if(localStorage.userData === undefined){
      return this.renderLogin()
    }else {
      return <Home logout={this.logout.bind(this)}
                   deets={localStorage.userData}
                   users={this.state.users}
                   friendReqs={this.state.friendReqs}
                   search={this.handleSearch.bind(this)}
                   videos={this.state.videos}
                   incoming={this.state.requests}
                   accept={this.acceptFriend.bind(this)}
              />
    }
  }

  render() {
    return (
      <div className='App'>
      {this.loginOrHome()}
      </div>
    )
  }
}

export default App;
