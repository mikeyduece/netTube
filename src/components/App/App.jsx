import './App.scss'
//Dependencies
import React from 'react'
import { useSelector } from "react-redux";

import GoogleAuth from "../Auth/GoogleAuth";
import Home from "../Home/Home";

const App = () => {
  const { isSignedIn } = useSelector(state => state.auth)

  const renderApp = () => {
    if (isSignedIn) {
      return <Home/>
    } else {
      return (
          <div className='app'>
            <h1 id='logo'>netTube</h1>
            <GoogleAuth/>
          </div>
      )
    }
  }

  return (
      <>
        { renderApp() }
      </>
  )
}


export default App

