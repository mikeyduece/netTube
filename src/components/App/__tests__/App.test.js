import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer';
import App from '../App'

it('should render the App component without crashing', function () {
  const component = renderer.create(
      <App/>
  )
  
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
