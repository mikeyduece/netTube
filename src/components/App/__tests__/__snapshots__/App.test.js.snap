// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `"App"`;

exports[`should render the App component without crashing 1`] = `"App"`;
